### PR TITLE
Initial CRD changes

### PR DESCRIPTION
**Pull request by aruniiird**
**Is it a bug or feature request?**
Neither bug or feature request, it is mainly required changes to start the project.

**What all things have been done to enable the changes?**

Following steps has been done to generate CRD files and a call in controller logic

1. Change the 'storagealerts_type.go' and add your needed data structures
2. Run operator-sdk add crd --api-version=alert.monstorak.org/v1alpha1 --kind=StorageAlerts
to generate CRDs.
3. Run operator-sdk generate openapi
Catch here is the json names of the type.go (here: storagealerts_type.go)
file should be matching with their corresponding variable names

Example
```
StorageAlert     AlertDetails     `json:"storageAlert,omitempty"`
StorageDashboard DashboardDetails `json:"storageDashboard,omitempty"`
```

4. If you have changed the controller logic (Reconcile function),
run operator-sdk generate k8s
_________________

Added the original crd design docs

Fixing issue: https://github.com/monstorak/monstorak/issues/6

Signed-off-by: aruniiird <arun.iiird@gmail.com>

### CODE REVIEW

**Review by umangachapagain — CHANGES_REQUESTED** [review]
1. Change the Title to CRD
2. Add Issue reference to Description
3. Validate Description
4. In CR file, use placeholders instead of general texts
5. Update String() function to use 'fmt' in cephalert_types.go

File: deploy/role.yaml [hunk 1]
```diff
@@ -50,5 +50,6 @@ rules:
   - alert.monstorak.org
   resources:
   - '*'
+  - cephalerts
```
Comment (shtripat) [inline deploy/role.yaml]: I feel we discussed earlier that this operator could be generic one we ideally should not use ceph/gluster etc at CRD defintiion level. IIUC this is generic portion, so suggesting this. Once we need storage system specific code, we can have packages defined accordingly.

**Comment by aruniiird** [comment]
@umangachapagain , @shtripat I have made the needed changes, please take a look...

**Comment by umangachapagain** [comment]
LGTM

File: deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: storagealerts.alerts.monstorak.org
+spec:
+  group: alerts.monstorak.org
+  names:
+    kind: StorageAlerts
+    listKind: StorageAlertsList
+    plural: storagealerts
+    singular: storagealerts
```
Comment (shtripat) [inline deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml:11]: plural and singular names, both same? Shouldn't we have singular as `storagealert` ?

Comment (aruniiird) [reply deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml:11]: Intentionally kept the plural, the thought process behind was, we are showing multiple alerts not a single alert.

Comment (shtripat) [reply deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml:11]: I feel its more represents a Storage alerts class so lets rename as `StorageAlertsClass` and then we will have proper singular as `storagealertclass` and plural as `storagealertclasses`. Sorry I am making too many changes, but makes sense I feel :)

Comment (aruniiird) [reply deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml:11]: @shtripat , as agreed, I will merge this PR and we will test our over all flow. We have a separate issue raised (and corresponding PR checked in) for the basic CRD design (issue:https://github.com/monstorak/monstorak/issues/6) . We will discuss the changes and finalize there. Once it is up to the mark, we can put the final changes to the code.

File: deploy/role.yaml (hunk 1 above)
Comment (aruniiird) [reply deploy/role.yaml]: Taken into account. Resolved.

File: deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml (hunk 2 above)
Comment (aruniiird) [reply deploy/crds/alerts_v1alpha1_storagealerts_crd.yaml:11]: PR tracking the issue: https://github.com/monstorak/monstorak/pull/19

**Pull request merged**